### ISSUE TITLE
Poor types after changes to useMeasure

### ISSUE THREAD

**Issue by coffeedoughnuts** [streamich/react-use#1222]
**What is the current behavior?**

After upgrading to 15, the generated types for `useMeasure` are:

```typescript
(() => UseMeasureResult) | (() => (Pick<DOMRectReadOnly, "left" | "right" | "bottom" | "top" | "x" | "y" | "width" | "height"> | (() => void))[])
```

which means that when you use it in the documented way:

```typescript
const [ref, { width, height }] = useMeasure();
```

you get type errors on the destructure because bother elements in the tuple can be functions.

I think this is because of the `useMeasureMock` which doesn't have explicitly typings.

**Steps to reproduce it and if possible a minimal demo of the problem. Your bug will get fixed much faster if we can run your code and it doesn't have extra dependencies other than `react-use`. Paste the link to your [JSFiddle](https://jsfiddle.net) or [CodeSandbox](https://codesandbox.io) example below:**

<img width="875" alt="Screenshot 2020-05-18 at 14 41 33" src="https://user-images.githubusercontent.com/4455835/82219851-c6fc2e00-9915-11ea-81db-0d5fbc9c5f82.png">

**What is the expected behavior?**

the destructure (as in the doc) works without type issues

**A little about versions:**  
- _OS_:  MacOS
- _Browser (vendor and version)_: Chrome
- _React_: 16.13.1
- _`react-use`_: 15.0.2
- _Did this worked in the previous package version?_ Yes, 14.2.0

**Comment by Svish** [comment]
Yep, this is the problem preventing me from upgrading to v15 too. Combined with the ref type being locked to `HTMLElement`. Should be fixed in my PR #1221.

**Comment by streamich** [comment]
Should be closed here: https://github.com/streamich/react-use/pull/1221
